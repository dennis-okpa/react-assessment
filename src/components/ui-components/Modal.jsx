import React from 'react';
import PropTypes from 'prop-types';

import {
  Panel,
  ActionPanel,
} from '@components/styled-components/Task';

const Modal = ({
  title, explanation, actionPanel, children
}) => (
  <Panel>
    <div>
      <h3>{title}</h3>
      <hr />
      <span>{explanation}</span>
    </div>
    {children}
    <ActionPanel>{actionPanel}</ActionPanel>
  </Panel>
);

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  explanation: PropTypes.node.isRequired,
  actionPanel: PropTypes.node.isRequired,
  children: PropTypes.node
};

Modal.defaultProps = {
  children: null
};

export default Modal;
