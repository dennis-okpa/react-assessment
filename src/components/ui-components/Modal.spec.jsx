import React from 'react';
import { render } from '@testing-library/react';
import Modal from './Modal';

describe('The Modal Component', () => {
  describe('when rendered', () => {
    let container;
    let getByText;
    let getByTestId;

    beforeEach(() => {
      ({ getByText, getByTestId } = render(
        <Modal
          title="mock-title"
          explanation="mock-explanation"
          actionPanel={(
            <div data-testid="mock-acton-panel" />
          )}
        >
          <div data-testid="mock-body" />
        </Modal>
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should display title', () => {
      expect(getByText('mock-title')).toBeInTheDocument();
    });

    it('should display explanation', () => {
      expect(getByText('mock-explanation')).toBeInTheDocument();
    });

    it('should display action panel', () => {
      expect(getByTestId('mock-acton-panel')).toBeInTheDocument();
    });

    it('should display content body', () => {
      expect(getByTestId('mock-body')).toBeInTheDocument();
    });
  });
});
