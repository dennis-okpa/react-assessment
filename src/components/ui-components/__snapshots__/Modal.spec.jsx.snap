// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Modal Component when rendered should match snapshot 1`] = `undefined`;
